src/Spryker/Zed/AppMerchant/Business/PaymentTransmissions/PaymentTransmissionsRequestExtender.php

```diff
@@ -122,4 +100,89 @@ protected function findMerchantForOrderItem(array $merchantTransfers, string $me
 
         return null;
     }
+
+    /**
+     * @param list<string> $merchantReferences
+     *
+     * @return array<\Generated\Shared\Transfer\MerchantTransfer>
+     */
+    protected function findMerchants(array $merchantReferences, string $tenantIdentifier): array
+    {
+        $merchantCriteriaTransfer = (new MerchantCriteriaTransfer())
+            ->setMerchantReferences($merchantReferences)
+            ->setTenantIdentifier($tenantIdentifier);
+
+        return $this->appMerchantRepository->findMerchants($merchantCriteriaTransfer);
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    protected function groupPaymentTransmissionItemsByOrderReferenceAndMerchantReference(
+        PaymentTransmissionsRequestTransfer $paymentTransmissionsRequestTransfer
+    ): array {
+        $merchantReferences = [];
+        $paymentTransmissionItemsGroupedByOrderReferenceAndMerchant = [];
+
+        foreach ($paymentTransmissionsRequestTransfer->getPaymentTransmissions() as $paymentTransmissionTransfer) {
+            foreach ($paymentTransmissionTransfer->getPaymentTransmissionItems() as $paymentTransmissionItemTransfer) {
+                if (!$paymentTransmissionItemTransfer->getMerchantReference()) {
+                    continue;
+                }
+
+                $orderReference = $paymentTransmissionItemTransfer->getOrderReference();
+                $merchantReference = $paymentTransmissionItemTransfer->getMerchantReference();
+
+                if (!isset($paymentTransmissionItemsGroupedByOrderReferenceAndMerchant[$orderReference])) {
+                    $paymentTransmissionItemsGroupedByOrderReferenceAndMerchant[$orderReference] = [];
+                }
+
+                if (!isset($paymentTransmissionItemsGroupedByOrderReferenceAndMerchant[$orderReference][$merchantReference])) {
+                    $paymentTransmissionItemsGroupedByOrderReferenceAndMerchant[$orderReference][$merchantReference] = [
+                        static::KEY_PAYMENT_TRANSMISSION_ITEMS => [],
+                        static::KEY_PAYMENT_TRANSMISSION => clone $paymentTransmissionTransfer,
+                    ];
+                }
+
+                $paymentTransmissionItemsGroupedByOrderReferenceAndMerchant[$orderReference][$merchantReference][static::KEY_PAYMENT_TRANSMISSION_ITEMS][] = $paymentTransmissionItemTransfer;
+                $merchantReferences[$merchantReference] = $merchantReference;
```

Comment:
looks interesting :)

is it to save one  `array_unique` call later?

Reply:
Yes)